src/components/ha-icon-next.ts

```diff
@@ -1,19 +1,18 @@
 import { mdiChevronLeft, mdiChevronRight } from "@mdi/js";
+import { type PropertyValues } from "lit";
 import { customElement } from "lit/decorators";
 import { HaSvgIcon } from "./ha-svg-icon";
 
 @customElement("ha-icon-next")
 export class HaIconNext extends HaSvgIcon {
-  public connectedCallback() {
-    super.connectedCallback();
+  path = mdiChevronRight;
 
-    // wait to check for direction since otherwise direction is wrong even though top level is RTL
-    setTimeout(() => {
-      this.path =
-        window.getComputedStyle(this).direction === "ltr"
-          ? mdiChevronRight
-          : mdiChevronLeft;
-    }, 100);
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+    this.path =
+      window.getComputedStyle(this).direction === "ltr"
+        ? mdiChevronRight
+        : mdiChevronLeft;
   }
 }
```

Comment:
I think we can make this a lot simpler and faster by just doing:

```
@customElement("ha-icon-next")
export class HaIconNext extends HaSvgIcon {
  @property() public path =
    document.dir === "ltr" ? mdiChevronRight : mdiChevronLeft;
}
```

Reply:
(We should however listen for changes in direction, so either lang update or `dir` attribute, but we dont do that now either)

Reply:
Problem is that the direction doesn't load at the start (the computed value is empty), which is why we do this in the first place.
I'm not sure how to listen to RTL changes - I did some research and couldn't find any good CSS or JS solutions, is there a way other than explicitly passing a direction to each component?

Reply:
> Problem is that the direction doesn't load at the start (the computed value is empty), which is why we do this in the first place.

That was when we where computing the styles, but my proposal just checks a property of the document

Reply:
Ah, I missed that part. I'll apply that.

Reply:
You missed the `@property() public` part of my suggestion 😉